matcher.go

```diff
@@ -86,141 +86,403 @@ func (m *Matcher) skipValue(token tokenType) error {
 	panic("unexpected value")
 }
 
-func (m *Matcher) resolveParam(in interface{}) FastVal {
-	if opVal, ok := in.(VarRef); ok {
-		panic(fmt.Sprintf("Cannot read %d", opVal.VarIdx))
+func (m *Matcher) literalFromSlot(slot SlotID) FastVal {
+	value := NewMissingFastVal()
+
+	savePos := m.tokens.Position()
+
+	slotInfo := m.slots[slot-1]
+	m.tokens.Seek(slotInfo.start)
+	token, tokenData, _ := m.tokens.Step()
+
+	if isLiteralToken(token) {
+		var parser fastLitParser
+		value = parser.Parse(token, tokenData)
 	}
-	if opValV, ok := in.(FastVal); ok {
-		return opValV
-	} else {
+
+	m.tokens.Seek(savePos)
+
+	return value
+}
+
+func (m *Matcher) resolveFunc(fn FuncRef, activeLit *FastVal) FastVal {
+	switch fn.FuncName {
+	case "mathRound":
```

Comment:
better make it a constant to ensure consistency

Reply:
I don't think we use this in more than one place, however I suspect you are right as perhaps we can implement this as a shared map between the compiler and matcher, such that you will get compiler errors for non-existent methods rather than errors at execution time.  Ideally it will be possible to guarantee that any generated Matcher is valid and will not error (except in the case of malformed input JSON).